### ISSUE TITLE
StatusCodeErrorHandlerService fails to handle server error

### ISSUE THREAD

**Issue by szilardd** [abpframework/abp#17857]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

After I upgraded the Angular frontend to version 7.4.0, when a 500 server error occurs for example with the following response

```js
HTTP/1.1 500 Internal Server Error
Content-Type: application/json; charset=utf-8

{
    "error": {
        "code": null,
        "message": "An internal error occurred during your request!",
        "details": null,
        "data": {
            "HelpLink.ProdName": "Microsoft SQL Server",
            "HelpLink.ProdVer": "15.00.4153",
            "HelpLink.EvtSrc": "MSSQLServer",
            "HelpLink.EvtID": "208",
            "HelpLink.BaseHelpUrl": "https://go.microsoft.com/fwlink",
            "HelpLink.LinkId": "20476"
        },
        "validationErrors": null
    }
}
```

Then a console error is logged in the frontend

```js
ERROR TypeError: Cannot destructure property 'status' of 'object null' as it is null.
    at StatusCodeErrorHandlerService.canHandle (abp-ng.theme.shared.mjs:2432:17)
    at abp-ng.theme.shared.mjs:2587:42
    at Array.find (<anonymous>)
    at ErrorHandler.handleError (abp-ng.theme.shared.mjs:2587:18)
    at Object.next (abp-ng.theme.shared.mjs:2578:36)
    at ConsumerObserver.next (Subscriber.js:91:33)
    at SafeSubscriber._next (Subscriber.js:60:26)
    at SafeSubscriber.next (Subscriber.js:31:18)
    at subscribe.innerSubscriber (switchMap.js:14:144)
    at OperatorSubscriber._next (OperatorSubscriber.js:13:21)
```

The error is thrown from [StatusCodeErrorHandlerService](https://github.com/abpframework/abp/blob/99806b7621fdc10cf2c6e6f4d994612656bda870/npm/ng-packs/packages/theme-shared/src/lib/services/status-code-error-handler.service.ts#L60)

Because this error handler fails, the generic error handler is triggered which shows this error popup:

![image](https://github.com/abpframework/abp/assets/968502/041f1bb4-7397-4f55-a84d-8d1069098877)

### Reproduction Steps

1. Rename `AbpUsers` table or do something else that triggers an error
2. Go to Administration -> Identity Management -> Users page
3. The error described above is logged to the console and the generic error popup is shown

### Regression?

The error didn't happen in v7.3. Seems to have been introduced in v7.4

### Version

7.4.0

### User Interface

Angular

### Database Provider

EF Core (Default)

### Tiered or separate authentication server

Tiered

### Operation System

Windows (Default)

**Comment by szilardd** [comment]
There's also a warning logged in the console which I think is related

```md
The localization source separator (::) not found.

warn@abp-ng.core.mjs:939
getLocalization@abp-ng.core.mjs:942
instant@abp-ng.core.mjs:912
transform@abp-ng.core.mjs:3415
pureFunction2Internal@core.mjs:23498
ɵɵpipeBind2@core.mjs:23718
_class4_div_0_h1_5_Template@abp-ng.theme.shared.mjs:159
```

I think it's because it expects a translation token, but the error message is already translated as `An internal error occurred during your request!`?

**Comment by Sinan997** [comment]
@szilardd we will fix this at 7.4 patch

for now u can use this as a workaround

in **`app.module.ts`**
```ts
import { ErrorHandler } from '@abp/ng.theme.shared';
@NgModule({
  imports: [...],
  declarations: [...],
  providers: [
        ... ,
        { provide: ErrorHandler, useExisting: CustomHandler }],
})
export class AppModule {}
```

**`CustomHandler`**
```ts
import { inject, Injectable, Injector } from '@angular/core';
import { HttpErrorResponse } from '@angular/common/http';
import { Observable, of, throwError } from 'rxjs';
import { catchError, filter, switchMap } from 'rxjs/operators';

import { HttpErrorReporterService } from '@abp/ng.core';
import { ConfirmationService, RouterErrorHandlerService, HTTP_ERROR_CONFIG, CUSTOM_ERROR_HANDLERS, HTTP_ERROR_HANDLER, CustomHttpErrorHandlerService, Confirmation, DEFAULT_ERROR_LOCALIZATIONS, DEFAULT_ERROR_MESSAGES, ErrorHandler } from '@abp/ng.theme.shared';

@Injectable({ providedIn: 'root' })
export class CustomHandler {
  protected readonly httpErrorReporter = inject(HttpErrorReporterService);
  protected readonly confirmationService = inject(ConfirmationService);
  protected readonly routerErrorHandlerService = inject(RouterErrorHandlerService);
  protected readonly httpErrorConfig = inject(HTTP_ERROR_CONFIG);
  protected readonly customErrorHandlers = inject(CUSTOM_ERROR_HANDLERS);
  protected readonly defaultHttpErrorHandler = (_, err: HttpErrorResponse) => throwError(() => err);
  protected readonly httpErrorHandler =
    inject(HTTP_ERROR_HANDLER, { optional: true }) || this.defaultHttpErrorHandler;

  constructor(protected injector: Injector) {
    this.listenToRestError();
    this.listenToRouterError();
  }

  protected listenToRouterError() {
    this.routerErrorHandlerService.listen();
  }

  protected listenToRestError() {
    this.httpErrorReporter.reporter$
      .pipe(filter(this.filterRestErrors), switchMap(this.executeErrorHandler))
      .subscribe(err => this.handleError(err));
  }

  protected executeErrorHandler = (error: HttpErrorResponse) => {
    const errHandler = this.httpErrorHandler(this.injector, error);
    const isObservable = errHandler instanceof Observable;

    return (isObservable ? errHandler : of(null)).pipe(
      catchError(err => {
        return of(err);
      }),
    );
  };

  protected sortHttpErrorHandlers(
    a: CustomHttpErrorHandlerService,
    b: CustomHttpErrorHandlerService,
  ) {
    return (b.priority || 0) - (a.priority || 0);
  }

  protected handleError(err: unknown) {
    if (this.customErrorHandlers && this.customErrorHandlers.length) {
      const canHandleService = this.customErrorHandlers
        .sort(this.sortHttpErrorHandlers)
        .find(service => service.canHandle(err));

      if (canHandleService) {
        canHandleService.execute();
        return;
      }
    }

    this.showError().subscribe();
  }

  protected showError(): Observable<Confirmation.Status> {
    const title = {
      key: DEFAULT_ERROR_LOCALIZATIONS.defaultError.title,
      defaultValue: DEFAULT_ERROR_MESSAGES.defaultError.title,
    };
    const message = {
      key: DEFAULT_ERROR_LOCALIZATIONS.defaultError.details,
      defaultValue: DEFAULT_ERROR_MESSAGES.defaultError.details,
    };
    return this.confirmationService.error(message, title, {
      hideCancelBtn: true,
      yesText: 'AbpAccount::Close',
    });
  }

  protected filterRestErrors = ({ status }: HttpErrorResponse): boolean => {
    if (typeof status !== 'number') return false;

    if (!this.httpErrorConfig || !this.httpErrorConfig.skipHandledErrorCodes) {
      return true;
    }

    return this.httpErrorConfig.skipHandledErrorCodes?.findIndex(code => code === status) < 0;
  };
}
```
